atomate/qchem/firetasks/critic2.py

```diff
@@ -0,0 +1,178 @@
+            molecule = fw_spec.get("prev_calc_molecule")
+        else:
+            molecule = self.get("molecule")
+        if molecule == None:
+            raise ValueError("No molecule passed and no prev_calc_molecule found in spec! Exiting...")
+
+        compress_at_end = False
+
+        if not self.get("testing"):
+
+            cube = self.get("cube_file")
+
+            if cube[-3:] == ".gz":
+                compress_at_end = True
+                decompress_file(cube)
+                cube = cube[:-3]
+
+            input_script = ["molecule "+cube]
+            input_script += ["load "+cube]
+            input_script += ["auto"]
+            input_script += ["CPREPORT CP.json"]
+            input_script += ["YT JSON YT.json"]
+            input_script += ["end"]
+            input_script += [""]
+            input_script = "\n".join(input_script)
+
+            with open('input_script.cri', 'w') as f:
+                f.write(input_script)
+            args = ["critic2", "input_script.cri"]
+
+            rs = subprocess.Popen(args,
+                                  stdin=subprocess.PIPE,
+                                  stdout=subprocess.PIPE,
+                                  stderr=subprocess.PIPE,
+                                  close_fds=True)
+
+            stdout, stderr = rs.communicate()
+            stdout = stdout.decode()
+
+            if stderr:
+                stderr = stderr.decode()
+                warnings.warn(stderr)
+                with open('stdout.cri', 'w') as f:
+                    f.write(stdout)
+                with open('stderr.cri', 'w') as f:
+                    f.write(stderr)
+
+            if rs.returncode != 0:
+                raise RuntimeError("critic2 exited with return code {}.".format(rs.returncode))
+
+        cp_loaded = loadfn("CP.json")
+        bohr_to_ang = 0.529177249
+
+        species = {}
+        for specie in cp_loaded["structure"]["species"]:
+            if specie["name"][1] == "_":
+                species[specie["id"]] = specie["name"][0]
+            else:
+                species[specie["id"]] = specie["name"]
+
+        atoms = []
+        centering_vector = cp_loaded["structure"]["molecule_centering_vector"]
+        for ii,atom in enumerate(cp_loaded["structure"]["nonequivalent_atoms"]):
+            specie = species[atom["species"]]
+            atoms.append(specie)
+            tmp = atom["cartesian_coordinates"]
+            coords = []
+            for jj,val in enumerate(tmp):
+                coords.append((val+centering_vector[jj])*bohr_to_ang)
+            if str(molecule[ii].specie) != specie:
+                raise RuntimeError("Atom ordering different!")
+            if molecule[ii].distance_from_point(coords) > 1*10**-5:
+                raise RuntimeError("Atom position "+str(ii)+" inconsistent!")
+
+        assert cp_loaded["critical_points"]["number_of_nonequivalent_cps"] == cp_loaded["critical_points"]["number_of_cell_cps"]
```

Comment:
Can you change this to:

```python
if (
    cp_loaded["critical_points"]["number_of_nonequivalent_cps"] != 
    cp_loaded["critical_points"]["number_of_cell_cps"]
):
    raise ValueError("error message")
```

As assertions will be ignored if python is run with optimisations.

Reply:
Changed in latest commit